packages/node-http-handler/src/node-http2-handler.spec.ts

```diff
@@ -121,55 +121,51 @@ describe("NodeHttp2Handler", () => {
-        expect(numRequests).toBe(2);
-        await assert.rejects(
-          nodeH2Handler.handle(req, {}),
-          UNEXPECTEDLY_CLOSED_REGEX,
-          "should be rejected promptly due to goaway frame"
-        );
-        expect(establishedConnections).toBe(3);
-        expect(numRequests).toBe(3);
-
-        // should be able to recover from goaway after reconnecting to a server
-        // that doesn't send goaway, and reuse the TCP connection (Http2Session)
-        shouldSendGoAway = false;
-        mockH2Server3.on("request", createResponseFunction(mockResponse));
-        await nodeH2Handler.handle(req, {});
-        const result = await nodeH2Handler.handle(req, {});
-        const resultReader = result.response.body;
-
-        // ...and validate that the mocked response is received
-        const responseBody = await new Promise((resolve) => {
-          const buffers = [];
-          resultReader.on("data", (chunk) => buffers.push(chunk));
-          resultReader.on("end", () => {
-            resolve(Buffer.concat(buffers).toString("utf8"));
-          });
+      const req = new HttpRequest({ ...getMockReqOptions(), port: port3 });
+      expect(establishedConnections).toBe(0);
+      expect(numRequests).toBe(0);
+      await assert.rejects(
+        nodeH2Handler.handle(req, {}),
+        UNEXPECTEDLY_CLOSED_REGEX,
+        "should be rejected promptly due to goaway frame"
+      );
+      expect(establishedConnections).toBe(1);
+      expect(numRequests).toBe(1);
+      await assert.rejects(
+        nodeH2Handler.handle(req, {}),
+        UNEXPECTEDLY_CLOSED_REGEX,
+        "should be rejected promptly due to goaway frame"
+      );
+      expect(establishedConnections).toBe(2);
+      expect(numRequests).toBe(2);
+      await assert.rejects(
+        nodeH2Handler.handle(req, {}),
+        UNEXPECTEDLY_CLOSED_REGEX,
+        "should be rejected promptly due to goaway frame"
+      );
+      expect(establishedConnections).toBe(3);
+      expect(numRequests).toBe(3);
+
+      // should be able to recover from goaway after reconnecting to a server
+      // that doesn't send goaway, and reuse the TCP connection (Http2Session)
+      shouldSendGoAway = false;
+      mockH2Server3.on("request", createResponseFunction(mockResponse));
+      await nodeH2Handler.handle(req, {});
+      const result = await nodeH2Handler.handle(req, {});
+      const resultReader = result.response.body;
+
+      // ...and validate that the mocked response is received
+      const responseBody = await new Promise((resolve) => {
+        const buffers = [];
+        resultReader.on("data", (chunk) => buffers.push(chunk));
+        resultReader.on("end", () => {
+          resolve(Buffer.concat(buffers).toString("utf8"));
         });
-        expect(responseBody).toBe("test");
-        expect(establishedConnections).toBe(4);
-        expect(numRequests).toBe(5);
-      } finally {
-        mockH2Server3.close();
```

Comment:
The intent of this check was to avoid making other tests fail if this test failed (if they ever reused the same port in the future) - it could waste time debugging if `"handles connections destroyed by servers"` below failed because it's also using port3 (or something even less straightforward).

Now that I think of it, checking for the mockH2Server in an after() handler instead and calling mockH2Server.close() may have been cleaner than this try{}finally{} approach. Thoughts?

Reply:
`mockH2Server.close()` is called in `afterAll` as mockH2Server is reused between tests.

If any additional servers are created by tests, they need to be closed within the test.
For example, `mockH2Server2` while testing new authority and `mockH2Server3` for goaway frames or connections destroyed by servers.